mypy/checker.py

```diff
@@ -67,7 +67,32 @@ def min_with_None_large(x: T, y: T) -> T:
 
 
 class Frame(Dict[Any, Type]):
-    pass
+    """Frame for the ConditionalTypeBinder.
+
+    Frames actually in the binder must have
+    binder.frames[i].parent == binder.frames[i-1].
+
+    options_on_return represents a list of possible frames that we
+    will incorporate next time we are the top frame (e.g. frames from
+    the end of each if/elif/else block).
+    """
+
+    changed = False
+
+    def __init__(self, parent: 'Frame' = None) -> None:
+        super().__init__()
+        self.parent = parent
+        self.options_on_return = []  # type: List[Frame]
```

Comment:
OK, I've moved all the members from Frame into ConditionalTypeBinder, so it's just a dict.